### PR TITLE
Add js bundler and add unit tests for class validators

### PR DESCRIPTION
**Pull request by Yuripetusko**
- Moved cli index into cli folder and outside of src
- Created ESM / CJS bundles, already tested in React and it works|
image

You can see how it is imported in react project here:
https://github.com/Yuripetusko/rmrk-ui-react/blob/main/src/components/rmrk-init.tsx

Also added extra unit tests

TODO:
- use webpack for esm/cjs bundling also instead of rollup

### CODE REVIEW

**Comment by Swader** [comment]
I don't understand what this does, can you dumb it down for me?

**Comment by Yuripetusko** [comment]
@Swader this transpiles and then bundles code into 2 versions. esm (for modenr node and has full tree shaping) and cjs (for modern-ish browsers iusing import/require and has partial tree shaking)

**Comment by Swader** [comment]
But what for exactly? Which problem does this solve?

**Comment by Yuripetusko** [comment]
`This is primarily needed to package library on npm/github later and use in external projects like I do already in rmrk-ui-react` and (something I need to add to TODO) still have all TS type definitions when importing rmrk-tools in react/svelte etc

**Comment by Yuripetusko** [comment]
Previously you were building using tsc and then manually bundling using browserify? Plus you were bundling polkadot.js libs all together in one browser bundle (no tree shaking etc)

This now in my opinion does a better job at this, minus browser bundle for now (the one you can include with <script> tag

**Comment by Yuripetusko** [comment]
This is how you can use rmrk-tools once it's bundled:

Install as npm package
https://github.com/Yuripetusko/rmrk-ui-react/blob/main/package.json#L21
And then import
https://github.com/Yuripetusko/rmrk-ui-react/blob/main/src/components/rmrk-init.tsx#L2

**Comment by Yuripetusko** [comment]
Basically a pre-requisit for writing rmrk UIs in vue/react/svelte or whatever

**Comment by Swader** [comment]
I see. Well, I only bundled *because* of browser support, no other reason since the imports can be pulled into react and stuff anyway, no? That's how Jaco does it in [this](https://github.com/polkadot-js/apps). So browser is a top priority for me, but I agree that it's too bloated as is. The reason is the WASM dependency in the keyring I think, that's not so easy to get rid of, but would be great if we could reduce the size. So tree shaking in the past didn't work, that's too sticky.

Where do you define what gets exported as part of `* as rmrk-tools`? I see your code directly uses web3enable, does that mean it just automatically enables all public exports from all the @polkadot packages, AND all public exports from rmrk? Or is there a specific definition file for those? (I'm new at this bundling stuff sorry)

**Comment by Swader** [comment]
NVM I'm an idiot, it's imported separately right there  🤦‍♂️

**Comment by Yuripetusko** [comment]
Added separate webpack build for browser bundle (the one you can import using <script tag> for true masochists ;) And also added unit tests for validating remark string
